splitio/sync/split.py

```diff
@@ -55,32 +55,38 @@ def _fetch_until(self, fetch_options, till=None):
         :return: last change number
         :rtype: int
         """
+        segment_list = []
         while True:  # Fetch until since==till
             change_number = self._split_storage.get_change_number()
             if change_number is None:
                 change_number = -1
             if till is not None and till < change_number:
                 # the passed till is less than change_number, no need to perform updates
-                return change_number
+                return change_number, segment_list
 
             try:
                 split_changes = self._api.fetch_splits(change_number, fetch_options)
             except APIException as exc:
                 _LOGGER.error('Exception raised while fetching splits')
                 _LOGGER.debug('Exception information: ', exc_info=True)
                 raise exc
-
+            
             for split in split_changes.get('splits', []):
                 if split['status'] == splits.Status.ACTIVE.value:
                     self._split_storage.put(splits.from_raw(split))
                 else:
                     self._split_storage.remove(split['name'])
-
+                for segment in self._split_storage.get_segment_names():
+                    _LOGGER.debug('Found segment: %s', segment)
+                    if not segment_sync.segment_exist_in_storage(segment):
+                        _LOGGER.debug('Segment %s does not exist, syncing.', segment)
+                        segment_list.append(segment)
+                
```

Comment:
i think we're asking for the whole list of segment names in a loop for every updated split.
we can do something along the lines of:

```
           for split in split_changes.get('splits', []):
                if split['status'] == splits.Status.ACTIVE.value:
                    parsed = splits.from_raw(split)
                    self._split_storage.put(parsed)
                    referenced_segments.update(set(parsed.get_segment_names()))
                else:
                    self._split_storage.remove(split['name'])
```

Reply:
Hi Martin, this logic is also used for on-going sync, so we need to only sync the new segments added to splits, and ignore existing segments in storage. That is why we need to check for the segment before we sync